grype/version/gemfile_version.go

```diff
@@ -0,0 +1,59 @@
+package version
+
+import (
+	"fmt"
+	"strings"
+)
+
+type gemfileVersion struct {
+	// keeping the raw version for transparency,
+	// but its value might change to fit semVer standards: https://semver.org/
+	raw    string
+	semVer *semanticVersion
+}
+
+// Gemfile.lock versions may have "{cpu}-{os}" or "{cpu}-{os}-{version}"
+// after the semvVer, for example, 12.2.1-alpha-x86_64-darwin-8, where `2.2.1-alpha`
+// is a valid and comparable semVer, and `x86_64-darwin-8` is not a semVer due to
+// the underscore. Also, we can't sort based on arch and OS in a way that make sense
+// for versions. SemVer is a characteristic of the code, not which arch OS it runs on.
+//
+// CPU is the most structured value present in gemfile.lock versions, we use it
+// to split the version info in half, the first half has semVer, and
+// the second half has arch and OS which we ignore.
+func extractSemVer(raw string) string {
+	lower := strings.ToLower(raw)
+
+	cpus := []string{"-x86", "-x86_64", "-universal", "-arm", "-armv5", "-armv6", "-armv7"}
```

Comment:
it would be ideal to also decouple the suffixes so you can specify a more generic list like `-arm\d*` so we don't need to updated the list as new architecture versions are introduced.

Reply:
I agree, by keeping just "arm" as a split token we can cover that case, and it is simpler than a regex.

Reply:
Right, I'm saying that this can be simplified down to
```diff 
-	cpus := []string{"-x86", "-x86_64", "-universal", "-arm", "-armv5", "-armv6", "-armv7"}
+	cpus := []string{"-x86", "-universal", "-arm"}
```